toml/decoder.py

```diff
@@ -702,17 +703,18 @@ def load_value(self, v, strictly_valid=True):
             elif v[0] == '+':
                 v = v[1:]
             v = v.replace('_', '')
-            if '.' in v or 'e' in v or 'E' in v:
-                if '.' in v and v.split('.', 1)[1] == '':
-                    raise ValueError("This float is missing digits after "
-                                     "the point")
-                if v[0] not in '0123456789':
-                    raise ValueError("This float doesn't have a leading "
-                                     "digit")
-                v = float(v)
-                itype = "float"
-            else:
-                v = int(v)
+            if 'x' not in v:
+                if '.' in v or 'e' in v or 'E' in v:
+                    if '.' in v and v.split('.', 1)[1] == '':
+                        raise ValueError("This float is missing digits after "
+                                         "the point")
+                    if v[0] not in '0123456789':
+                        raise ValueError("This float doesn't have a leading "
+                                         "digit")
+                    v = float(v)
+                    itype = "float"
+            if itype == "int":
```

Comment:
Where does itype get set to `"int"` ?

Reply:
Line 697
But it will be changed to "float" if the number is, well, a float.
Base 10 numbers are fine without this but hex, binary and octal need to be converted to base 10 with the ```v = int(v, 0)``` thing in the next line.